rose/pre_process/default_trains.py

```diff
@@ -233,210 +259,289 @@ def set_sprinter_slt_train(time, velocities, start_coord):
-        bogie.calculate_total_n_dof()
-
-        # setup wheels per bogie
-        bogie.wheels = [Wheel() for _ in range(len(bogie.wheel_distances))]
-        for wheel in bogie.wheels:
-            wheel.mass = 2.5e3
+    cargo_train.carts = [Cart() for _ in range(nb_carts)]
+    cargo_train.cart_distances = [start_coord - i * cargo_length for i in range(nb_carts)]
+
+    for cart in cargo_train.carts:
+        cart.bogie_distances = [-3.745, 3.745]
+        cart.inertia = 1558125/2
+        cart.mass = 65e3/2
+        cart.stiffness = 24e6
+        cart.damping = 71e3
+        cart.length = cargo_length
+        cart.calculate_total_n_dof()
+
+        # setup bogies per cart
+        cart.bogies = [Bogie() for _ in range(len(cart.bogie_distances))]
+        for bogie in cart.bogies:
+            bogie.wheel_distances = [-0.91, 0.91]
+            bogie.mass = 9333.3/2
+            bogie.inertia = 9.3e3/2
+            bogie.stiffness = 2.232e6
+            bogie.damping = 36.7e3
+            bogie.length = 1.82
+            bogie.calculate_total_n_dof()
+
+            # setup wheels per bogie
+            bogie.wheels = [Wheel() for _ in range(len(bogie.wheel_distances))]
+            for wheel in bogie.wheels:
+                wheel.mass = 2.5e3
 
     return cargo_train
 
+def build_cargo_train(time: np.ndarray, velocities: np.ndarray, start_coord: float, locomotive_type: TrainType,
+                      wagon_type: TrainType, nb_locomotives=1, nb_wagons=1):
+    """
+    Builds a cargo train model with locomotives and wagons
+
+    :param time: all time steps
+    :param velocities: velocity of the train
+    :param start_coord: initial coordinate of the cart
+    :param locomotive: type of locomotive
+    :param wagon: type of wagon
+    :param nb_locomotives: number of locomotives (default 1)
+    :param nb_wagons: number of wagons (default 1)
+
+    :return: TrainModel
+    """
+
+    # define locomotive
+    if locomotive_type == TrainType.TRAXX:
+        cargo_train = set_traxx_locomotive(time, velocities, start_coord, nb_carts=nb_locomotives)
+    elif locomotive_type == TrainType.BR189:
+        cargo_train = set_br189_locomotive(time, velocities, start_coord, nb_carts=nb_locomotives)
+    else:
+        raise ValueError(f"Locomotive {locomotive_type} not defined")
+
+    # define wagon
+    if wagon_type == TrainType.CARGO_TAPPS:
+        wagon_train = set_cargo_TAPPS_train(time, velocities, start_coord, nb_carts=1)
+    elif wagon_type == TrainType.CARGO_FALNS5:
+        wagon_train = set_cargo_FALNS5_train(time, velocities, start_coord, nb_carts=1)
+    elif wagon_type == TrainType.CARGO_SGNS:
+        wagon_train = set_cargo_SGNS_train(time, velocities, start_coord, nb_carts=1)
+    else:
+        raise ValueError(f"Wagon {wagon_type} not defined")
+
+    wagon_length = wagon_train.carts[0].length
+    cargo_train.cart_distances.extend([start_coord - cargo_train.carts[0].length - wagon_length * i for i in range(nb_wagons)])
```

Comment:
this is still not correct as the start coordinate starts at the middle of the carts/ locomotives, thus the distance between the first wagon and cart should be half locomotive + half wagon